hack/lib/util/golang.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+#
+# This library holds golang related utility functions.
+
+# Ensure the go tool exists and is a viable version.
+os::golang::verify_go_version() {
+  if [[ -z "$(which go)" ]]; then
+    cat <<EOF
+Can't find 'go' in PATH, please fix and retry.
+See http://golang.org/doc/install for installation instructions.
+EOF
+    return 1
+  fi
+
+  local go_version
+  go_version=($(go version))
+  if [[ "${go_version[2]}" < "go1.6" && "${go_version[2]}" != "devel" ]]; then
```

Comment:
I'm not sure this check will be sufficient -- we don't currently support `go1.7`, and on the test machines we *don't* want to be running any of these steps with `go1.7` until we choose to, since otherwise an overnight `go` install could break all of our jobs. I think what we could do is provide an override flag for users like you who are doing things locally, but the argument before when I proposed that was we don't want people locally seeing everything pass then going to Jenkins and seeing failures, since we don't have a guarantee of backwards compat for stuff like `gofmt` or `govet`.